### ISSUE TITLE
After vMotion, Couldn't access a container on bridge.

### ISSUE THREAD

**Issue by kakakikikeke** [vmware/vic#7493]
# Version
* VIC v1.3.0-rc1-0-4c17da5 (from docker info)
* ESXi 6.5
* vCenter 6.5.0 build 4598948

# Details
I have two hosts in one cluster.  
A vch was created on hostA (called s26) with the command below.

* Creating a vch command

```
curl \
-X POST \
-u 'user:pass' \
'http://localhost:18080/container/target/192.168.1.2/datacenter/datacenter-21/vch?thumbprint=F9%3A03%3A2C%3A25%3A45%3A07%3ACE%3A24%3A4F%3A14%3A70%3A95%3A7E%3AB4%3A68%3A87%3AB1%3A2E%3AC1%3AEC' \
-H 'content-type: application/json' \
-d @- << EOF
{
  "name": "test_vch",
  "compute": {
    "resource": {
      "name": "Cluster001"
    },
    "cpu": {
      "limit" :{
        "units": "MHz",
        "value": 512
      },
      "reservation" :{
        "units": "MHz",
        "value": 0
      }
    },
    "memory": {
      "limit": {
        "units": "MiB",
        "value": 512
      },
      "reservation" :{
        "units": "MiB",
        "value": 0
      }
    }
  },
  "storage": {
    "volume_stores": [{
      "datastore": "vol1/test",
      "label": "default"
    }],
    "image_stores": ["vol1"]
  },
  "network": {
    "bridge": {
      "ip_range": "172.17.0.0/12",
      "port_group": {
        "name": "bridge001"
      }
    },
    "public": {
      "port_group": {
        "name": "glo001"
      },
      "gateway": {
        "address": "xxx.xxx.xxx.1"
      },
      "static": "xxx.xxx.xxx.xxx/25",
      "nameservers": ["8.8.8.8"]
    },
    "management": {
      "port_group": {
        "name": "bridge002"
      },
      "gateway": {
        "address": "10.1.0.1",
        "routing_destinations": ["172.22.0.0/16"]
      },
      "static": "10.1.0.4/24",
      "nameservers": ["8.8.8.8"]
    }
  }
}
EOF
```

Viewing on vSphere Client.
![created_vch](https://user-images.githubusercontent.com/1751050/37260338-64720918-25d5-11e8-90a3-74d9c6e7789f.jpg)

The secret part is masked.

Next, I created a container with docker run.  
Then the container was created on host A (s 26) in the same way.

```
docker -H xxx.xxx.xxx.xxx:2375 run -d -p 80:80 nginx
```

```
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                     NAMES
114b4b5c0595        nginx               "nginx -g daemon off;"   35 minutes ago      Up 21 minutes       xxx.xxx.xxx.xxx:80->80/tcp   sharp_perlman
```

![created_container1](https://user-images.githubusercontent.com/1751050/37260582-748721d2-25d8-11e8-839e-c433f49b9bbc.jpg)

The container is nginx.  
In this state you can get html by accessing nginx.

And I moved this container to another host (hostB called s30).
![created_container2](https://user-images.githubusercontent.com/1751050/37260586-7b9bb55a-25d8-11e8-8552-386a1d1f2793.jpg)

As a result, I couldn't access nginx.  

```
Failed to connect to xxx.xxx.xxx.xxx port 80: No route to host
```

Returned to the original host (s30 -> s26) will be normal.

# Questions
* Is there a rule that vch and container must always be on the same host?
* Whether this behavior is service specification or not (Is it due to instant cloning?)

**Comment by AngieCris** [comment]
@anchal-agrawal Anchal have you seen this before? Do you happen to have input on it?

**Comment by AngieCris** [comment]
@kakakikikeke Can you attach the container logs from the VCH?

**Comment by kakakikikeke** [comment]
[vch.zip](https://github.com/vmware/vic/files/1801379/vch.zip)

**Comment by anchal-agrawal** [comment]
@AngieCris I have not.

> And I moved this container to another host

This part stands out to me. If a vMotion was performed manually using the vSphere client, it would be an out-of-band operation and it's possible we don't account for it yet. Per my understanding, a "standard" vMotion from vSphere should trigger an event and be handled by our backend to avoid this situation.

@kakakikikeke 

> Is there a rule that vch and container must always be on the same host?

No; containerVMs and VCHs are placed according to DRS when it is enabled.

**Comment by mdubya66** [comment]
If at all possible you should upgrade to 1.3.1. There are quite a few fixes in the engine https://github.com/vmware/vic/compare/v1.3.0-rc1...v1.3.1

We'd also need to see the vsphere logs for this problem. I'd suggest opening a case with VMWare support once you've upgraded to v.1.3.1

The comment regarding out of band vMotion is probably not relevant. We do test that scenario https://github.com/vmware/vic/blob/master/tests/manual-test-cases/Group13-vMotion/13-2-vMotion-Container.robot

cc @hickeng

**Comment by kakakikikeke** [comment]
@anchal-agrawal 

> If a vMotion was performed manually using the vSphere client, it would be an out-of-band operation and it's possible we don't account for it yet.

Yes, I did manually vmotion. (Select a container -> light click -> Migrate ...)  
Does it mean that manual vmotion is not supported?

@mdubya66 OK, I will upgrade to v1.3.1 and retry it.

**Comment by kakakikikeke** [comment]
After upgrading to v1.3.1, I tried the same vMotion manually but it was a failure.  

I think that manual vMotion using vSphere web client probably uses MigrateVM_Task API.  
So I tried calling RelocateVM_Task using the SOAP API, but this also failed the result.

Perhaps it may not work except DRS's vMotion.

**Comment by hickeng** [comment]
@kakakikikeke It should be irrelvent whether it's vMotion manually or vMotion via DRS.

There is no requirement that the VCH endpointVM and the containerVMs be on the same host, however there _is_ a requirement that there be a Distributed Port Group configured between the hosts for the bridge network and that it be L2 isolated so there are no IP address collisions between different VCHs that use the same ranges for the bridge networks.

Given your description of the issue my working hypothesis is that the bridge network is misconfigured. This can be confirmed/refuted by:
a. manually migrating the containerVM to the second host, confirming that communication fails, then migrating the VCH endpointVM to the second host as well and confirming that the communication now succeeds
b. create two container VMs on the same host, and use the second one to ping the first (should succeed). Then vMotion the first containerVM to the second host and retry the ping which will fail if I'm correct.

**Comment by kakakikikeke** [comment]
@hickeng I resolved it.

I changed the bridge network to another distributed port group instead of vic-bridge-001 and it worked fine.  
Probably I think that something in the setting of vic-bridge-001 is bad.  
I will review the setting again (Even I tried to set the VLAN ID, to change the CIDR of the bridge, to check the setting of the physical NIC, but in particular I could not find a strange point).

Everyone, thank you for helping me.
